deepchem/hyper/gaussian_process.py

```diff
@@ -5,238 +5,249 @@
-      self,
-      params_dict,
-      train_dataset,
-      valid_dataset,
-      output_transformers,
-      metric,
-      direction=True,
-      n_features=1024,
-      n_tasks=1,
-      max_iter=20,
-      search_range=4,
-      hp_invalid_list=[
-          'seed', 'nb_epoch', 'penalty_type', 'dropouts', 'bypass_dropouts',
-          'n_pair_feat', 'fit_transformers', 'min_child_weight',
-          'max_delta_step', 'subsample', 'colsample_bylevel',
-          'colsample_bytree', 'reg_alpha', 'reg_lambda', 'scale_pos_weight',
-          'base_score'
-      ],
-      log_file='GPhypersearch.log'):
-    """Perform hyperparams search using a gaussian process assumption
-
-    params_dict include single-valued parameters being optimized,
-    which should only contain int, float and list of int(float)
-
-    parameters with names in hp_invalid_list will not be changed.
-
-    For Molnet models, self.model_class is model name in string,
-    params_dict = dc.molnet.preset_hyper_parameters.hps[self.model_class]
+  def hyperparam_search(self,
+                        params_dict,
+                        train_dataset,
+                        valid_dataset,
+                        transformers,
+                        metric,
+                        use_max=True,
+                        logdir=None,
+                        max_iter=20,
+                        search_range=4,
+                        logfile=None):
+    """Perform hyperparameter search using a gaussian process.
 
     Parameters
     ----------
     params_dict: dict
-      dict including parameters and their initial values
-      parameters not suitable for optimization can be added to hp_invalid_list
-    train_dataset: dc.data.Dataset struct
+      Maps hyperparameter names (strings) to possible parameter
+      values. The semantics of this list are different than for
+      `GridHyperparamOpt`. `params_dict[hp]` must map to an int/float,
+      which is used as the center of a search with radius
+      `search_range`.
+    train_dataset: `dc.data.Dataset`
       dataset used for training
-    valid_dataset: dc.data.Dataset struct
+    valid_dataset: `dc.data.Dataset`
       dataset used for validation(optimization on valid scores)
-    output_transformers: list of dc.trans.Transformer
+    transformers: list[dc.trans.Transformer]
       transformers for evaluation
-    metric: list of dc.metrics.Metric
+    metric: `dc.metrics.Metric`
       metric used for evaluation
-    direction: bool
+    use_max: bool, (default True)
       maximization(True) or minimization(False)
-    n_features: int
-      number of input features
-    n_tasks: int
-      number of tasks
-    max_iter: int
+    logdir: str, optional
+      The directory in which to store created models. If not set, will
+      use a temporary directory.
+    max_iter: int, (default 20)
       number of optimization trials
-    search_range: int(float)
+    search_range: int(float) (default 4)

```

Comment:
Is it reasonable to assume this will always have the same value for all hyperparameters?

Reply:
Good point. I've fixed it so that it can accept different search ranges for each parameter